src/Microsoft.Health.Blob/Registration/BlobClientRegistrationExtensions.cs

```diff
@@ -133,7 +133,7 @@ public static IServiceCollection AddBlobServiceClient(this IServiceCollection se
                     {
                         clientBuilder = builder
                             .AddBlobServiceClient(new Uri(options.ConnectionString))
-                            .WithCredential(new DefaultAzureCredential(new DefaultAzureCredentialOptions { ManagedIdentityClientId = options.ClientId }));
+                            .WithCredential(new DefaultAzureCredential(options.Credentials));
```

Comment:
Like in my other comment, should we always use the credentials and instead just check for managed identity to determine which overload of `AddBlobServiceClient` to call?

Reply:
But you're right that we do need to use the different overloads. It's very non-obvious though lol. The string overload uses the key-value pair connection strings like `DefaultEndpointsProtocol=https;AccountName=storagesample;AccountKey=<account-key>`, while `Uri` overload takes the base URL for all azure storage account services.

Reply:
Updated!